src/opencmiss/utils/zinc/field.py

```diff
@@ -411,6 +412,114 @@ def get_or_create_field_coordinates(fieldmodule : Fieldmodule, name="coordinates
+    New field has component names: "fibre angle", "imbrication angle", "sheet angle".
+    New field is not managed by default.
+
+    :param fieldmodule:  Zinc fieldmodule to find or create field in.
+    :param name:  Name of field to find or create.
+    :param components_count: Number of components of field, from 1 to 3.
+    :param managed: Managed state of field if created here.
+    :return: Zinc FieldFiniteElement
+    """
+    assert 1 <= components_count <= 3
+    with ZincCacheChanges(fieldmodule):
+        fibres = create_field_finite_element(fieldmodule, name, components_count,
+                                             component_names=["fibre angle", "imbrication angle", "sheet angle"],
+                                             managed=managed)
+        fibres.setCoordinateSystemType(Field.COORDINATE_SYSTEM_TYPE_FIBRE)
+    return fibres
+
+
+def get_or_create_field_fibres(fieldmodule : Fieldmodule, name="fibres", components_count=3) -> FieldFiniteElement:
+    """
+    Finds or creates a finite element fibre field.
+    New field has component names: "fibre angle", "imbrication angle", "sheet angle".
+
+    :param fieldmodule:  Zinc fieldmodule to find or create field in.
+    :param name:  Name of field to find or create.
+    :param components_count: Number of components of field, from 1 to 3.
+    :return: Zinc FieldFiniteElement
+    """
+    assert 1 <= components_count <= 3
+    if field_exists(fieldmodule, name, components_count=components_count):
+        fibres = fieldmodule.findFieldByName(name).castFiniteElement()
+        if fibres.getCoordinateSystemType() == Field.COORDINATE_SYSTEM_TYPE_FIBRE:
+            return fibres
+    return create_field_fibres(fieldmodule, name, components_count, managed=True)
+
+
+def create_field_group(fieldmodule : Fieldmodule, name : str, managed=False) -> FieldGroup:
+    """
+    Finds or creates a Group field of the supplied name.
+
+    :param fieldmodule:  Zinc fieldmodule to find or create field in.
+    :param name:  Name of field to find or create.
+    :param managed: Managed state of field if created here.
+    :return: Zinc FieldGroup.
+    """
+    with ZincCacheChanges(fieldmodule):
+        group = fieldmodule.createFieldGroup()
+        group.setName(name)
+        group.setManaged(managed)
+    return group
+
+
+def get_or_create_field_group(fieldmodule : Fieldmodule, name : str) -> FieldGroup:
+    """
+    Finds or creates a Group field of the supplied name.
+    New field is managed.
+
+    :param fieldmodule:  Zinc fieldmodule to find or create field in.
+    :param name:  Name of field to find or create.
+    :return: Zinc FieldGroup.
+    """
+    field = fieldmodule.findFieldByName(name)
+    if field.isValid():
+        group = field.castGroup()
+        if group:
+            return group
+    return create_field_group(fieldmodule, name, managed=True)
+
+def group_get_or_create_nodeset_group(group : FieldGroup, nodeset : Nodeset) -> NodesetGroup:
```

Comment:
I think this should be `get_or_create_group_nodeset_group` of `get_or_create_nodeset_group`.

Reply:
Decided it's better to create the FieldNodeGroup instead; caller can call .getNodesetGroup() on it. Note FieldNodeGroup can be created either in or out of a group but most common use is likely to be in the group.